js/extension/components/FeatureInfoButton.jsx

```diff
@@ -0,0 +1,114 @@
+
+/**
+ * Copyright 2015, GeoSolutions Sas.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import '../assets/infoButton.css';
+
+import assign from 'object-assign';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Glyphicon} from 'react-bootstrap';
+import Button from '@mapstore/components/misc/Button';
+
+import Dialog from '@mapstore/components/misc/Dialog';
+
+/**
+ * A button to show a simple information window.
+ * Component's properies:
+ *  - id: {string}            custom identifier for this component
+ *  - title: {string|element} title of the window shown in its header
+ *  - body: {string|element}  content of the window
+ *  - style: {object}         a css-like hash to define the style on the component
+ *  - glyphicon: {string}     bootstrap glyphicon name
+ *  - text: {string|element}  text content for the button
+ *  - btnSize: {string}       bootstrap button size ('large', 'small', 'xsmall')
+ *  - btnType: {string}       'normal' to use a standard bootstrap button;
+ *                            'image'  to use a custom image as button
+ *
+ * Note: the button will not be never empty, it will show at least the text (default or custom)
+ */
+class FeatureInfoButton extends React.Component {
+    static propTypes = {
+        id: PropTypes.string,
+        image: PropTypes.string,
+        title: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        body: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        style: PropTypes.object,
+        className: PropTypes.string,
+        glyphicon: PropTypes.string,
+        text: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        help: PropTypes.oneOfType([PropTypes.string, PropTypes.element]),
+        hiddenText: PropTypes.bool,
+        btnSize: PropTypes.oneOf(['large', 'medium', 'small', 'xsmall']),
+        btnType: PropTypes.oneOf(['normal', 'image']),
+        modalOptions: PropTypes.object,
+        closeGlyph: PropTypes.string
+    };
+
+    static defaultProps = {
+        id: "mapstore-infobutton",
+        title: "Info",
+        body: "",
+        style: undefined,
+        glyphicon: undefined,
+        text: "Info",
+        hiddenText: false,
+        btnSize: 'medium',
+        btnType: 'normal',
+        modalOptions: {},
+        closeGlyph: ""
```

Comment:
If that's only a featureinfobutton, I guess this list of props can be widely simplified.